### PR TITLE
Convert rest docstrings to Google docstrings.

### PR DESCRIPTION
**Pull request by tgamblin**
Follow on to #3920.

- Sometimes you need something mindless to do.
- Sometimes it can be helpful and satisfy your OCD, as well.

### CODE REVIEW

File: var/spack/repos/builtin.mock/packages/flake8/package.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ class Flake8(Package):
     All of these violations are exceptions that we allow in
     package.py files."""
 
-    # Used to tell whether or not the package has been modified
+    # Used to tell whether or not the package has been unmodified
```
Comment (adamjstewart) [inline var/spack/repos/builtin.mock/packages/flake8/package.py]: I think modified sounds better here. I had the same thing too but I ended up changing it.

File: lib/spack/llnl/util/filesystem.py [hunk 2]
```diff
@@ -585,24 +585,23 @@ def __str__(self):
...
-    :type args: str or collections.Sequence
-    :param bool shared: if True searches for shared libraries,
+    Args:
+        args (str or list of str): Library name(s) to search for
+        shared (bool): searches for shared libraries if True
 
-    :returns: The libraries that have been found
-    :rtype: LibraryList
+    Returns:
+        LibaryList: The libraries that have been found
```
Comment (alalazo) [inline lib/spack/llnl/util/filesystem.py]: Lib**r**aryList

File: lib/spack/llnl/util/filesystem.py [hunk 3]
```diff
@@ -585,24 +585,23 @@ def __str__(self):
 
 
 def find_system_libraries(args, shared=True):
-    """Searches the usual system library locations for the libraries
-    specified in args.
+    """Searches the system library locations libraries.
```
Comment (alalazo) [inline lib/spack/llnl/util/filesystem.py]: I think something is missing here...

File: lib/spack/spack/cmd/url.py [hunk 4]
```diff
@@ -260,13 +261,15 @@ def print_name_and_version(url):
...
-    :param set urls: List of URLs that have already been added
-    :param url: A URL to potentially add to ``urls`` depending on ``args``
-    :type url: str or None
-    :param spack.package.PackageBase pkg: The Spack package
-    :returns: The updated ``urls`` list
-    :rtype: set
+    Args:
+        args (argparse.Namespace): The arguments given to ``spack url list``
+        urls (set): List of URLs that have already been added
+        ur (str or None): A URL to potentially add to ``urls`` depending on
```
Comment (alalazo) [inline lib/spack/spack/cmd/url.py]: ur**l**

File: var/spack/repos/builtin.mock/packages/flake8/package.py (hunk 1 above)
Comment (tgamblin) [inline var/spack/repos/builtin.mock/packages/flake8/package.py]: Not sure how that happened...

File: lib/spack/llnl/util/filesystem.py (hunk 3 above)
Comment (tgamblin) [inline lib/spack/llnl/util/filesystem.py]: First line (summary) in a docstring is supposed to be one line -- any suggestions?

Comment (tgamblin) [inline lib/spack/llnl/util/filesystem.py]: (Aside from correcting the obvious miswording, I mean :grimace:)

Comment (alalazo) [inline lib/spack/llnl/util/filesystem.py]: `Searches the usual system library locations for args` ?

Comment (adamjstewart) [inline lib/spack/llnl/util/filesystem.py]: > First line (summary) in a docstring is supposed to be one line

Is this a rule of thumb or a hard requirement? Guess I've never heard it before. Makes sense though.

Comment (tgamblin) [inline lib/spack/llnl/util/filesystem.py]: @adamjstewart: it's from [PEP 257](https://www.python.org/dev/peps/pep-0257/):

> Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description. The summary line may be used by automatic indexing tools; it is important that it fits on one line and is separated from the rest of the docstring by a blank line.

**Pull request merged**